ui/components/src/model/legend.ts

```diff
@@ -14,11 +14,14 @@
 import { MouseEventHandler } from 'react';
 
 export const legendPositions = ['Bottom', 'Right'] as const;
-
 export type LegendPositions = (typeof legendPositions)[number];
 
+export const legendModes = ['List', 'Table'] as const;
+export type LegendMode = (typeof legendModes)[number];
+
 export interface LegendOptions {
   position: LegendPositions;
+  mode?: LegendMode;
```

Comment:
Making this optional with a default, so its introduction isn't a breaking change.